src/server/report/test.js

```diff
@@ -142,6 +295,77 @@ class Test extends LitElement {
 		};
 		return html`<div class="pill-box">${items.map(i => renderItem(i))}</div>`;
 	}
+	_renderTabButtons(browsers, selectedBrowser, testData) {
+
+		const onKeyDown = (e) => {
+			let focusOn;
+			switch (e.key) {
+				case 'ArrowRight':
+					focusOn = e.target.nextElementSibling || e.target.parentNode.firstElementChild;
+					break;
+				case 'ArrowLeft':
+					focusOn = e.target.previousElementSibling || e.target.parentNode.lastElementChild;
+					break;
+				case 'Home':
+					focusOn = e.target.parentNode.firstElementChild;
+					break;
+				case 'End':
+					focusOn = e.target.parentNode.lastElementChild;
+					break;
+			}
+			if (focusOn) focusOn.focus();
+		};
+
+		const renderTabButton = (browser, index) => {
+			const result = testData.results.find(r => r.name === browser);
+			const selected = (browser === selectedBrowser.name);
+			const status = result.passed ? 'passed' : 'failed';
+			const onClick = () => {
+				return () => this._selectedBrowserIndex = index;
+			};
+			const statusClass = {
+				pass: result.passed,
+				error: !result.passed
+			};
+			return html`
+				<button
+					aria-controls="tabpanel-${browser}"
+					aria-selected="${selected ? 'true' : 'false'}"
+					id="tab-${browser}"
+					role="tab"
+					tabindex="${selected ? '0' : '-1'}"
+					type="button"
+					@click="${onClick()}">
+						<span>${browser} <span class="${classMap(statusClass)}">(${status})</span></span>
+						${selected ? html`<div class="tab-selected-indicator"></div>` : nothing}
+				</button>`;
```

Comment:
More of a question than a suggestion. I got into the habit of formatting like this because I found it easier to follow the markup, and easy _enough_ to then find and follow the javascript within/around it, which overall made it easier to find and read than, for example, trying to scan for that `<div>` in there.

Admittedly this looks pretty terrible in the github comment UI, though, and maybe it's just a messy example anyway.
```suggestion
				<button
					aria-controls="tabpanel-${browser}"
					aria-selected="${selected ? 'true' : 'false'}"
					id="tab-${browser}"
					role="tab"
					tabindex="${selected ? '0' : '-1'}"
					type="button"
					@click="${onClick()}">
						<span>
							${browser} <span class="${classMap(statusClass)}">(${status})</span>
						</span>${selected ? html`
						<div class="tab-selected-indicator"></div>` : nothing}
				</button>`;
```

Reply:
Sure why not!